Compatibility Shader Fails to Compile When Depth Buffer Used Inside Function

### Tested versions

- Reproducible in 4.4.1.stable.official.49a5bc7b6

### System information

Godot v4.4.1.stable - Ubuntu 24.04.2 LTS 24.04 on X11 - X11 display driver, Multi-window, 1 monitor - OpenGL 3 (Compatibility) - NVIDIA GeForce RTX 4060 (nvidia; 570.124.06) - 13th Gen Intel(R) Core(TM) i9-13900KS (32 threads)

### Issue description

When using the compatibility renderer, a shader that reads the depth texture from inside a function called by fragment(), will not compile.

Error:
```
E 0:00:00:516   _display_error_with_code: SceneShaderGLES3: Fragment shader compilation failed:
0(524) : error C1503: undefined variable "depth_buffer"

  <C++ Source>  drivers/gles3/shader_gles3.cpp:265 @ _display_error_with_code()
```

### Steps to reproduce

Create a compatibility renderer project, create a mesh with the shader
```gdshader
shader_type spatial;
render_mode unshaded, fog_disabled;

uniform sampler2D depth_texture : hint_depth_texture, filter_nearest;

float get_depth() {
	return texture(depth_texture, vec2(0.0, 0.0)).r;
}

void vertex() {
	POSITION = vec4(VERTEX.xy, 1.0, 1.0);
}

void fragment() {
	ALBEDO = vec3(get_depth());
}
```
Run the project

### Minimal reproduction project (MRP)

[shader_error.zip](https://github.com/user-attachments/files/21740875/shader_error.zip)

--- a/drivers/gles3/shaders/scene.glsl
+++ b/drivers/gles3/shaders/scene.glsl
@@ -1022,12 +1022,6 @@ uniform highp mat4 world_transform;
 uniform highp uint instance_offset;
 uniform highp uint model_flags;
 
-/* clang-format off */
-
-#GLOBALS
-
-/* clang-format on */
-
 #define LIGHT_BAKE_DISABLED 0u
 #define LIGHT_BAKE_STATIC 1u
 #define LIGHT_BAKE_DYNAMIC 2u
@@ -1267,6 +1261,12 @@ layout(location = 3) out vec4 emission_output_buffer;
 layout(location = 0) out vec4 frag_color;
 
 #endif // !RENDER_MATERIAL
+
+/* clang-format off */
+
+#GLOBALS
+
+/* clang-format on */
 
 vec3 F0(float metallic, float specular, vec3 albedo) {
 	float dielectric = 0.16 * specular * specular;

